common/dynamicconfig/shared_constants.go

```diff
@@ -24,4 +24,27 @@
 
 package dynamicconfig
 
-const DefaultNumTaskQueuePartitions = 4
+import "go.temporal.io/server/common/primitives"
+
+var defaultNumTaskQueuePartitions = []ConstrainedValue{
+	// The per-ns worker task queue in all namespaces should only have one partition, since
+	// we'll only run one worker per task queue.
+	{
+		Constraints: map[string]any{
+			TaskQueueName.String(): primitives.PerNSWorkerTaskQueue,
+		},
+		Value: 1,
+	},
+	// All task queues in the system namespace are relatively low-throughput, they only need
+	// one partition.
+	{
+		Constraints: map[string]any{
+			Namespace.String(): primitives.SystemLocalNamespace,
+		},
+		Value: 1,
+	},
```

Comment:
This may potentially cause some system workflow to stuck if a task is sitting in other partition.

Reply:
ah, darn. I'll revert this until we have a solution for that